### ISSUE TITLE
Lint against `String::from_raw_parts`

### ISSUE THREAD

**Issue by robertbastian** [rust-lang/rust-clippy#14293]
### What it does

Lints against usages of `String::from_raw_parts`. This should be done in two steps, `Vec::from_raw_parts` and `String::from_utf8_unchecked`.

### Advantage

- `String::from_raw_parts` has conflated safety requirements that should be justified in two steps:
  - `Vec::from_raw_parts` has complicated language-level safety requirements that have to be manually checked and are insta-UB (?) if violated
  - `String::from_utf8_unchecked` has a relatively trivial library-level safety requirement, that could be checked by using `String::from_utf8` instead

### Example

```rust
<code>
```

Could be written as:

```rust
<code>
```

**Comment by samueltardieu** [comment]
Note that you can already warn about or deny using `String::from_raw_parts()` by using the `clippy::disallowed_methods` lint. You can even indicate the reason for disallowing it.

**Comment by robertbastian** [comment]
I'm aware. I'm not the only one with this concern: https://github.com/rust-lang/libs-team/issues/167#issuecomment-1399622667

**Comment by robertbastian** [comment]
Yes a new lint, could be pedantic, style, restriction, suspicious, idk.
